libr/bin/p/bin_bflt.c

```diff
@@ -21,10 +21,10 @@ static RList *entries(RBinFile *bf) {
 	return ret;
 }
 
-static void __patch_reloc(RBuffer *buf, ut32 addr_to_patch, ut32 data_offset) {
+static void __patch_reloc(RIOBind *iob, ut32 addr_to_patch, ut32 data_offset) {
 	ut8 val[4] = { 0 };
 	r_write_le32 (val, data_offset);
-	r_buf_write_at (buf, addr_to_patch, (void *) val, sizeof (val));
+	iob->overlay_write_at (iob->io, addr_to_patch, val, sizeof (val));
```

Comment:
my main concern on this is that your changes makes it impossible to test (without building a separate branch) with the current layered io cache. so i would like to make this configurable at runtime to choose if u want the relocs to be written on the overlay or in the buffer (actually rbufwrite() was silently calling the io api under the hood)